packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx

```diff
@@ -0,0 +1,121 @@
+import {type ListenEvent, type ListenOptions} from '@sanity/client'
+import {useCallback, useEffect, useMemo, useState} from 'react'
+import {catchError, of} from 'rxjs'
+import {
+  type BundleDocument,
+  DEFAULT_STUDIO_CLIENT_OPTIONS,
+  getBundleSlug,
+  useBundles,
+  useClient,
+} from 'sanity'
+
+export interface DocumentPerspectiveProps {
+  documentId: string
+}
+
+export interface DocumentPerspectiveState {
+  data: BundleDocument[] | null
+  error?: Error
+}
+
+const LISTEN_OPTIONS: ListenOptions = {
+  events: ['welcome', 'mutation', 'reconnect'],
+  includeResult: true,
+  visibility: 'query',
+}
+
+/**
+ * Fetches the document versions for a given document
+ * @param props - document Id of the document (might include bundle slug)
+ * @returns - data: document versions, loading, errors
+ * @hidden
+ * @beta
+ */
+export function useDocumentVersions(props: DocumentPerspectiveProps): DocumentPerspectiveState {
+  const {documentId} = props
+
+  const [state, setState] = useState<BundleDocument[] | null>(null)
+
+  const [error, setError] = useState<Error | undefined>(undefined)
+  const client = useClient(DEFAULT_STUDIO_CLIENT_OPTIONS)
+  const {data: bundles} = useBundles()
+  const pureDocumentId =
+    documentId.indexOf('.') > 0
```

Comment:
There's already a `getPublishedId()` in this package, could we reuse that instead?

Reply:
getPublishedId only accounts for the drafts.(id) structure.
Would we be ok with changing this? I just thought about this way because I was concerned about changing those things 🤔

Reply:
That's a good point. Could we add a new function (`getBundlePublishedId()`?) that includes the logic for bundle ids for now? Id like for us to have a central place to keep this logic instead of spreading it around.

Reply:
Ahhh I actually see that @juice49  added something for this already in the getPublishedId
<img width="732" alt="image" src="https://github.com/user-attachments/assets/7cc1bb55-69c0-45af-9ad7-39dedcc79b8a">

I'm thinking about moving this to a `getbundlePublishedId`, thoughts?

Reply:
Seems to have caused a test failure 🤔

Reply:
Just noticed, forgot to add the mock!
Should be fine now (just pushed the update)